src/wrappers/contract_wrappers/debt_token_wrapper.ts

```diff
@@ -530,10 +531,20 @@ export class DebtTokenContract extends BaseContract {
                 txData,
                 self.safeTransferFrom.estimateGasAsync.bind(self, _from, _to, _tokenId, _data),
             );
-            const txHash = await promisify<string>(
-                self.web3ContractInstance.safeTransferFrom,
-                self.web3ContractInstance,
-            )(_from, _to, _tokenId, _data, txDataWithDefaults);
+
+            // Since truffle does not play nice with overloaded functions, we have to
+            //      1. Figure out which overloaded function we want to call based on
+            //          the presence or absence of data parameter
+            //      2. Manually construct the transaction and send it via a web3 sendTransaction call.
+            const txHash = await TransactionUtils.sendRawTransaction(
+                self.web3,
+                self.web3ContractInstance,
+                "safeTransferFrom",
+                `address,address,uint256${_data.length > 0 ? ",bytes" : ""}`,
```

Comment:
why is this interpolation necessary?

Reply:
can't we just always specify data and default to `""` -- this is what happens on the backend anyway.

Reply:
Fair enough, fixing.

Reply:
Actually -- I take that back -- constructing the raw transaction in our TransactionUtils is problematic when we pass in a blank parameter.  Having a lot of trouble getting this to work with this.  Unless this is a sticking point for you, going to leave this as is.